neuralmonkey/decoders/decoder.py

```diff
@@ -398,14 +402,20 @@ def _visualize_attention(self):
                 collections=["summary_val_plots"],
                 max_images=256)
 
-    def feed_dict(self, dataset, train=False):
+    def feed_dict(self, dataset: Dataset, train: bool=False) -> FeedDict:
         """Populate the feed dictionary for the decoder object
 
         Arguments:
             dataset: The dataset to use for the decoder.
             train: Boolean flag, telling whether this is a training run
         """
-        sentences = dataset.get_series(self.data_id, allow_none=True)
+        sentences = cast(Iterable[List[str]],
```

Comment:
Právě že nestačilo. Ta vocabulary žere List[List[str]], aby vrátila tensor a asi je fajn, že je to tam napsáno. Že series v datasetu je Iterable[Any] je taky asi v pořádku, ale  i když z toho udělám List[Any], tak si to pořád potom stěžuje, že Any není List[str], tak se to musí takhe o castovat.

Trochu jsem doufal, že ten cast aspoň zkontroluje typ za runtimu, ale nedělá nic:
```
In [4]: type(cast(int, "bflm"))
Out[4]: str
```

Zkusím to přepsat nějak přehledněji. Tohle je výsledek mnoha iterací, kdy jsem byl nakonec rád, že to funguje.